api/types/src/convert.rs

```diff
@@ -212,24 +224,34 @@ impl<'a, R: MoveResolver + ?Sized> MoveConverter<'a, R> {
         access_path: AccessPath,
         op: WriteOp,
     ) -> Result<WriteSetChange> {
+        let sk = StateKey::AccessPath(access_path);
+        let state_key_hash = sk.hash().to_hex_literal();
+        let access_path = match sk {
+            StateKey::AccessPath(access_path) => access_path,
+            _ => unreachable!(),
+        };
```

Comment:
this is horribly uncomfortable code

Reply:
agreed...

Reply:
I mean... you do have a result here, so use anyhow?

Reply:
it can never hit the unreachable though